dhis2/dhis2-core
# fix: clear file resource data value with empty value/file [DHIS2-12735]

Deleting a data value of type FILE should remove the associated file resource on deletion.
This should work for individual updates with null value or deletions as well as same scenario on data value set import.

### Manual Testing
* Create a data element of value type file (resource). In this example it has the ID `yBdVC3WCqdW`.
* Create a data value with file, e.g. `POST /api/dataValues/file?de=yBdVC3WCqdW&pe=202301&ou=ImspTQPwCqd&co=HllvX50cXC0` with header `Content-Type` = `multipart/form-data` and add a multi-part body with name `file` of some image for example.
* You can confirm the data value exist using `GET /api/dataValues?de=yBdVC3WCqdW&pe=202301&ou=ImspTQPwCqd&co=HllvX50cXC0` .
* The shown value is the ID of the file resource that can be checked using `GET /api/fileResources/<uid>`.

No the value and the associated file resource can be deleted in 4 ways.

1. update individual data value with null value: `POST /api/dataValues?de=<uid>&pe=<uid>&ou=<uid>&co=<uid>`
2. update individual data value with empty file: `POST /api/dataValues/file?de=<uid>&pe=<uid>&ou=<uid>&co=<uid>&file=
3. update via import: `POST /api/dataValueSets` with JSON body
```json
{"dataValues": [{
  "dataElement":"yBdVC3WCqdW",
  "period":"202301",
  "orgUnit":"ImspTQPwCqd",
  "categoryOptionCombo":"HllvX50cXC0",
  "deleted": true,
  "value": null
}]}
```
4. delete via import: `POST /api/dataValueSets?importStrategy=DELETE` with JSON body
```json
{"dataValues": [{
  "dataElement":"yBdVC3WCqdW",
  "period":"202301",
  "orgUnit":"ImspTQPwCqd",
  "categoryOptionCombo":"HllvX50cXC0"
}]}
```
